### ISSUE TITLE
Error with dribble after ball touch (Opta)

### ISSUE THREAD

**Issue by Zapotnik** [ML-KULeuven/socceraction#519]
Hi,

There is a problem with "dribbles" after converting Opta data into SPADL actions.
When the dribble event follows a ball touch event, it gives a wrong result.
Ball touch events don't have endX and endY coordinates. To be accurate, endX and endY coordinates are the same as start_x and start_y ones.
So the dribble event starts with the coordinates of the ball touch and ends with the beginning of the following event.

-- 

Is it voluntary that dribbles are interrupted by take ons ? Because a player often starts his run before a take on.

Thanks a lot for your work.

**Comment by probberechts** [comment]
Can you share a small sample of data that illustrates the problem?

**Comment by Zapotnik** [comment]
[test_data.csv](https://github.com/ML-KULeuven/socceraction/files/11323570/test_data.csv)

**Comment by probberechts** [comment]
Do you have any idea what actually happened here? Your example is not a straightforward action sequence. It describes that a player touched the ball after a a pass, then nothing happens for five seconds (which is quite long) and then the ball is suddenly 20 meters further.

**Comment by Zapotnik** [comment]
Yes, I have. That's why I gave you that example.
A bit of context this time : it's the first goal of Khvicha Kvaratskhelia againts Monza (21.08.2022). A run from the left and a shot from distance after a touch (ball touch) by Zielinski.
So, in the opta data, you have a ball touch (81.6, 48.1, 86.1, 48.1) and then a shot (76.2, 62, 100, 45.5). There is no coordinate for the beginning of the run.
In the SPADL actions, you have a dribble (86.1, 48.1, 76.2, 62) and the shot. So, the beginning of the run is the ball touch and the end is the shot. But it's a mistake.
This time, you have the data (Opta and SPADL) and a simple pitch with the wrong run (black) and the good one (red) that I created myself for the example.
[Kvara_but_events_and_SPADL.csv](https://github.com/ML-KULeuven/socceraction/files/11413393/Kvara_but_events_and_SPADL.csv)
![Kava_course](https://user-images.githubusercontent.com/124366981/236646273-65665221-9ef7-4516-8f42-a8ae373a074c.png)
